packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -587,12 +589,14 @@ class IOSDevice extends Device {
       timer.cancel();
       if (localUri == null) {
         await iosDeployDebugger?.stopAndDumpBacktrace();
+        await dispose();
         return LaunchResult.failed();
       }
       return LaunchResult.succeeded(vmServiceUri: localUri);
     } on ProcessException catch (e) {
       await iosDeployDebugger?.stopAndDumpBacktrace();
       _logger.printError(e.message);
+      await dispose();
       return LaunchResult.failed();
     } finally {
```

Comment:
any reason not to just call `dispose()` from the `finally` block?

Reply:
We don't want it to dispose when it launches successfully

Reply:
ahh yeah, good point :)